### PR TITLE
Revert "zfs: unlock for 5.15"

### PR DESCRIPTION
**Pull request by TredwellGit**
###### Motivation for this change
This reverts commit 86e057aa5b6b44aa20b65ed194ec359e897bd89d.

###### Things done
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/zfs/default.nix [hunk 1]
```diff
@@ -215,8 +215,8 @@ in {
   # to be adapted
   zfsStable = common {
     # check the release notes for compatible kernels
-    kernelCompatible = kernel.kernelAtLeast "3.10" && kernel.kernelOlder "5.16";
-    latestCompatibleLinuxPackages = linuxPackages_5_15;
+    kernelCompatible = kernel.kernelAtLeast "3.10" && kernel.kernelOlder "5.15";
+    latestCompatibleLinuxPackages = linuxPackages_5_14;
```
Comment (Mic92) [inline pkgs/os-specific/linux/zfs/default.nix:219]: What is this for? I don't see how it is used.

**Comment by Mic92** [comment]
> After having my zfs partition borked from the SEEK_HOLE issue, I'm apprehensive about pushing the bounds of zfs. I would like to be more on the side of caution. Especially since data integrity is the first priority for anything.

That was a zfs bug and not a compatibility issue but a zfs bug: https://github.com/openzfs/zfs/issues/11900

File: pkgs/os-specific/linux/zfs/default.nix (hunk 1 above)
Comment (TredwellGit) [reply pkgs/os-specific/linux/zfs/default.nix:219]: https://github.com/NixOS/nixpkgs/pull/126980

**Comment by jonringer** [comment]
> > After having my zfs partition borked from the SEEK_HOLE issue, I'm apprehensive about pushing the bounds of zfs. I would like to be more on the side of caution. Especially since data integrity is the first priority for anything.
> 
> That was a zfs bug and not a compatibility issue: [openzfs/zfs#11900](https://github.com/openzfs/zfs/issues/11900)

Although I absolutely agree, it still makes me more risk adverse.

**Comment by jansol** [comment]
You win some, you lose some I guess.

This in combination with #147446 means that I woke up the other day to not getting a graphical login screen. Turns out that thanks to 5.14 being EOL I'm now stuck on 5.10, where amdgpu does not yet have any support for my card at all (RDNA2).

**Comment by Mic92** [comment]
Well. You can just apply https://github.com/Mic92/nixpkgs/commit/b31e0be77366618eab2a3f7a0648ec1786504732
I used for some weeks now without issues.

**Comment by KenMacD** [comment]
@TredwellGit 2.1.2 is out, and lists 5.15 support :pleading_face:

**Comment by TredwellGit** [comment]
https://github.com/NixOS/nixpkgs/pull/150923

**Pull request merged**